### DISCUSSION TITLE
Is there a way to handle SnackbarMessage timer's duration end?

### DISCUSSION THREAD

**Discussion by hcalxhcalxhcalx** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2793] [Q&A] [answered]
Hello. Is it possible to handle/execute custom code when SnackbarMessage expires - that is, when user didn't click ActionButton, timer runs out and SnackbarMessage becomes inactive?

**Comment by Keboo** [comment, marked as answer]
At present there is not really anything directly built-in for this, though you could build out enough tracking for it. To get the expiration you could watch the `Message` property on the `Snackbar` itself to know when a message is being removed (though you would need to track the previous value yourself). You would then also need to track if the user clicked the action button.
